050-unidirectional-topic-operator.md

```diff
@@ -0,0 +1,801 @@
+</table>
+
+### Emitted Kube `Events`
+
+Let's consider all the cases in which the BTO emits `Event` resources to the Kube apiserver:
+
+* For errors when accessing the topic store. 
+  For UTO there is no topic store, so this cannot happen.
+* For errors when creating partitions. 
+  For UTO ApiExceptions from the Admin client will be reported via `Ready` condition.
+  For other errors application logging and metrics are considered sufficient.
+* For errors when changing the configuration of topics in Kafka
+  For UTO ApiExceptions from the Admin client will be reported via `Ready` condition.
+  For other errors application logging and metrics are considered sufficient.
+* For errors when processing a ZooKeeper znode watch.
+  UTO doesn't use znode watches, so this cannot happen.
+* On attempts to change a `KafkaTopic`'s `spec.topicName`.
+  For UTO this will be reported via the status only.
+* On detection of conflicting changes between Kube and Kafka
+  For UTO such conflicts cannot happen.
+* On attempts to decrease the number of partitions.
+  For UTO this will be reported via the status only.
+
+Thus, the UTO will emit no events.
+Any software which consumed BTO events will not be compatible with UTO.
+
+## Migration
+
+To migrate, existing BTO users would have to:
+
+1. Review how they're using the BTO and whether they require bidirectionality
+   1. If bidirectional support is required by their usage then the UTO cannot be used. 
+2. Undeploy the BTO (they can retain their existing `KafkaTopics`)
+3. Deploy the UTO (reconfiguring their `Kafka` CR, or raw `Deployment` in the standalone case). 
+4. Some reconfiguration of pod resources and JVM options may also be required. 
+
+### Disadvantages
+
+#### Use of finalizers
+
+Using finalizers on `KafkaTopics` means that the operator has to be running (to remove the finalizer) before the resources will actually be deleted.
+That can prevent other resources from being removed (for example the `Namespace` containing the `KafkaTopics`).
+In these cases it is possible to remove the finalizer manually (e.g. via `kubectl edit`), which will then allow the apiserver to remove any other resources.
+
+#### Compatibility with other tooling
+
+This approach is not easily compatible with other topic management tooling. This includes things like the `kafka-configs.sh` and other Apache Kafka scripts as well as any other tooling which uses the `Admin` client, including applications themselves.
+
+<!--Even using the "scheduled auto creation" option, changes done via the admin server would be reverted by the next timed reconciliation. -->
+
+In the particular case of an "Admin server", to provide CLI or UI functionality in a Strimzi-aware way it _might_ be possible to use wrap `Admin` (decorator pattern), or write it using some abstraction over `Admin` which could also support operations on `KafkaTopics` for those interactions which would otherwise result in conflicting sources of truth.
```

Comment:
There is no _Admin server_ in Strimzi. I think this should be removed or be more generic. The Admin server was removed / archived long time ago with the UI.